trnsysGUI/mainWindow.py

```diff
@@ -577,11 +600,14 @@ def closeEvent(self, e):
             if ret == _qtw.QMessageBox.Cancel:
                 e.ignore()
             elif ret == _qtw.QMessageBox.Close:
+                UserSettings.setRecentProjectJsonPath(self.editor.diagramPath)
                 e.accept()
             elif ret == _qtw.QMessageBox.Save:
                 self.editor.save()
+                UserSettings.setRecentProjectJsonPath(self.editor.diagramPath)
                 e.accept()
         else:
```

Comment:
Could we make the else into an early return?
Or just reverse the case (if not, then bigger set of cases).
Early returns would make this easier to understand.